cmd/stack_select.go

```diff
@@ -86,5 +97,13 @@ func newStackSelectCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVarP(
 		&stack, "stack", "s", "",
 		"The name of the stack to select")
+	cmd.PersistentFlags().BoolVarP(
+		&create, "create", "c", false,
+		"If selected stack does not exist, create it")
+	cmd.PersistentFlags().StringVar(
```

Comment:
In the spirit of validating user input, at the start of the command, we should check if `!create && secretsProvider != ""`, and if so report an error. As `pulumi stack --secrets-provider foo` makes no sense, and we shouldn't silently ignore that flag, IMHO.